### PR TITLE
WHFHRI-524: Upgrade to go 1.17

### PR DESCRIPTION
**Pull request by dan-baxter**
Signed-off-by: Dan Baxter <dan.baxter240@gmail.com>

## PR Requirements:
 - [ ] The code must be reviewed by at least one Maintainer
 - [x] JIRA Ticket: https://jira.wh-sdlc.watson-health.ibm.com/browse/WHFHRI-524
 - [x] README updates

### CODE REVIEW

File: src/go.mod [hunk 1]
```diff
@@ -11,25 +11,41 @@ require (
...
-	github.com/mattn/go-colorable v0.1.11 // indirect
 	github.com/newrelic/go-agent/v3 v3.15.0
 	github.com/newrelic/go-agent/v3/integrations/nrecho-v4 v1.0.2
 	github.com/peterbourgon/ff/v3 v3.1.2
-	github.com/pquerna/cachecontrol v0.1.0 // indirect
 	github.com/sirupsen/logrus v1.8.1
 	github.com/stretchr/testify v1.7.0
+)
+
+require (
```
Comment (David-N-Perkins) [inline src/go.mod:21]: This looks strange. Why are there two `require` blocks? Are all the new indirect entries due to forcing library upgrades?

File: docker/Dockerfile [hunk 2]
```diff
@@ -8,7 +8,7 @@ RUN microdnf update -y && \
     microdnf -y install make tar gzip gcc && \
     microdnf clean all
 
-RUN curl -f https://dl.google.com/go/go1.15.15.linux-amd64.tar.gz -o /tmp/golang.tar.gz && \
+RUN curl -f https://dl.google.com/go/go1.17.3.linux-amd64.tar.gz -o /tmp/golang.tar.gz && \
```
Comment (David-N-Perkins) [inline docker/Dockerfile:11]: I wonder if there's a way to always get the latest `1.17` version, so we automatically get patches.

File: src/go.mod (hunk 1 above)
Comment (dan-baxter) [reply src/go.mod:21]: Umm I'm not sure to be honest. When I upgraded to 1.17 and tried to build the docker image, I got errors saying I needed to run go mod tidy on the go.mod file, and these were just the automatic modifications from that. After that the docker image built. Should I have tried to update the go.mod file in another way that might have required fewer changes?

**Review by FRicci618 — APPROVED** [review]

File: docker/Dockerfile (hunk 2 above)
Comment (alisakotliarova) [reply docker/Dockerfile:11]: some bootleg pseudocode of an idea
```
for (x = 100; x > 2; x--) {
  try {
    RUN `curl -f https://dl.google.com/go/go1.17.{$x}.linux-amd64.tar.gz -o /tmp/golang.tar.gz`
    break;
  }
}
```

**Review by alisakotliarova — APPROVED** [review]

File: src/go.mod (hunk 1 above)
Comment (David-N-Perkins) [reply src/go.mod:21]: I updated locally and also got this change. Looks like all the `//indirect` lines were moved to a second `require` block. And when I tried to manually merge them, building separated them again. Must be a new convention.

File: docker/Dockerfile (hunk 2 above)
Comment (David-N-Perkins) [reply docker/Dockerfile:11]: I checked the website and don't see a way to get the latest, so I added it to our periodic 'check for updates' ticket.

**Review by David-N-Perkins — APPROVED** [review]

**Pull request merged**